consul/base.py

```diff
@@ -1409,17 +1474,47 @@ def node(self,
                 '/v1/catalog/node/%s' % node,
                 params=params)
 
+        def _service(self,
+                     internal_uri,
+                     index=None,
+                     wait=None,
+                     tag=None,
+                     consistency=None,
+                     dc=None,
+                     near=None,
+                     token=None,
+                     node_meta=None):
+            params = []
+            dc = dc or self.agent.dc
+            if dc:
```

Comment:
Could be factorized

`def append_params(params, key, value):
   ... 
`
      
Then
`append_params(params, 'dc', dc)`
`append_params(params, 'tag', tag)`
`append_params(params, 'index', index)`
`...`

Reply:
Yes, I just want to avoid doing such things only here while it's not done like this everywhere else.

Reply:
Fair enough